redact/v4/redact_requests.py

```diff
@@ -201,12 +201,20 @@ def _stream_output_to_file(
             finally:
                 if finished:
                     file_name = Path(retrieve_file_name(headers=response.headers))
-                    anonymized_path = Path(file.parent).joinpath(
-                        f"{file.stem}{file_name.suffix}"
+                    log.debug(f"getting headers file type suffix {file_name}")
+                    anonymized_path = (
+                        Path(file).parent / f"{file.stem}{file_name.suffix}"
+                    )
+
+                    target_file.rename(anonymized_path)
+                    log.debug(
+                        f"temp file {target_file} has been renamed into {anonymized_path}"
                     )
-                    os.rename(temp_file.name, str(anonymized_path))
+                    return anonymized_path
                 else:
-                    os.unlink(temp_file.name)
+                    target_file.unlink(missing_ok=True)
+
+                    raise RuntimeError(f"failed to download the file {file}")
```

Comment:
Please create a custom Error inside `redact/errors.py` which is more meaning full. Something like:
```
class FileDownloadError(Exception):
    """Raised when a file cannot be downloaded from the backend server."""
    def __init__(self, message):
        super().__init__(message)
```

Reply:
added it